x-pack/solutions/security/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/transforms.ts

```diff
@@ -140,3 +140,22 @@ const getDocumentEntryFromLegacyKbEntry = (
   };
   return documentEntry;
 };
+
+export const transformFieldNamesToSourceScheme = (fields: string[]) => {
+  return fields.map((f) => {
+    switch (f) {
+      case 'createdAt':
+        return 'created_at';
+      case 'updateddAt':
+        return 'updated_at';
+      case 'createdBy':
+        return 'created_by';
+      case 'updatedBy':
+        return 'updated_by';
+      case 'timestamp':
+        return '@timestamp';
```

Comment:
https://www.youtube.com/watch?v=7W0EO6C2sFg

I think it's fine to do snakecase, but we do have some `timestamp` fields that do not use the `@`, for example the [`timestamp` field in `Message`](https://github.com/elastic/kibana/blob/8c36b938368183c4183e8dcd6ae9e18089f71e38/x-pack/platform/packages/shared/kbn-elastic-assistant-common/impl/schemas/conversations/common_attributes.gen.ts#L242). I don't think we should be adjusting every `timestamp` field for this rason.

Reply:
Agree, I will drop timestamp and just use `_.snakeCase` from lodash